apps/website/components/templates/create-gate/schema.ts

```diff
@@ -233,39 +247,51 @@ export const quizDataSchema = z.object({
 });
 
 export const taskQuizSchema = z.object({
-  title: z.string().min(2),
-  description: z.string().min(2),
+  title: z.string().min(2, 'Quiz title must contain at least 2 character(s)'),
+  description: z
+    .string()
+    .min(2, 'Quiz description must contain at least 2 character(s)'),
   task_type: z.literal('quiz'),
   task_data: quizDataSchema,
 });
 
 export const taskHoldTokenSchema = z.object({
-  title: z.string().min(2),
-  description: z.string().min(2),
+  title: z
+    .string()
+    .min(2, 'Hold Token title must contain at least 2 character(s)'),
+  description: z
+    .string()
+    .min(2, 'The description must contain at least 2 character(s)'),
   task_type: z.literal('token_hold'),
   task_data: holdTokenTaskDataSchema,
 });
 
 export const taskSelfVerifySchema = z.object({
-  title: z.string().min(2),
-  description: z.string().min(2),
+  title: z.string().min(2, 'The title must contain at least 2 character(s)'),
+  description: z
+    .string()
+    .min(2, 'The description must contain at least 2 character(s)'),
   task_type: z.literal('self_verify'),
   task_data: fileTaskDataSchema,
 });
 
 export const taskSnapshotSchema = z.object({
-  title: z.string().min(2),
-  description: z.string().min(2),
+  title: z.string().min(2, 'The title must contain at least 2 character(s)'),
+  description: z
+    .string()
+    .min(2, 'The description must contain at least 2 character(s)'),
   task_type: z.literal('snapshot'),
   task_data: snapshotTaskDataSchema,
 });
 
 export const createGateSchema = z.object({
-  title: z.string().min(2),
-  categories: z.array(z.string()),
-  description: z.string().min(2),
+  title: z.string().min(2, 'The title must contain at least 2 character(s)'),
+  categories: z.array(z.string().min(1, 'Please select atleast 1 category')),
```

Comment:
You're checking it the wrong way. We technically want *at least* one category added to the array, however what you're checking is if the string which is added to array has at least one character.

Since all the categories have way more than 1 character, that's not a relevant check. 

```suggestion
  categories: z.array(z.string()).min(1, 'Please select at least 1 category'),
```

Reply:
Doing Changes